.github/workflows/publish-docker-image.yml

```diff
@@ -0,0 +1,30 @@
+name: Publish image to DockerHub
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
```

Comment:
Why do we need QEMU?

Reply:
Because we have multi-arch images to build, and QEMU is the way we have to emulate a different arch than the Action runner host. There is more explanation in the docker action https://github.com/docker/setup-qemu-action if you're interested :)